src/khan.ts

```diff
@@ -0,0 +1,133 @@
+import net from 'net'
+import { Cell, jam, dwim, Noun, enjs } from '@urbit/nockjs'
+import { newtDecode, newtEncode } from './bytes'
+
+// returns a js Noun Object with the structure of a %fyrd
+function Fyrd(desk: string, thread: string, mark_out: string, mark_in: string, atom: string) {
+  return dwim([
+    0,
+    'fyrd',
+    desk,
+    thread,
+    mark_out,
+    mark_in,
+    atom
+  ]);
+}
+
+// returns a js Fyrd, whose trailing atom is
+// some hoon code to be run as a thread
+function Thread (hoon: string) {
+  return Fyrd('base', 'khan-eval', 'noun', 'ted-eval', hoon);
+}
+
+interface ThreadResponse {
```

Comment:
the need for this makes me think we might want to rethink the approach of parseThreadResponse() in a TS context. Maybe its fine as is, maybe this is code smell though.